invictusd0a/inn-api
# #165369465 Add JWT authentication

###  What does this PR do?

This PR add JWT Authentication. When a user signs up via the registration endpoint, they should receive a JWT required on all subsequent calls to all other endpoints that require authentication.

### **How to test?**
Use postman to test

Signup with `http://127.0.0.1:8000/api/v1/auth/signup/` and Login with `http://127.0.0.1:8000/api/v1/auth/login/` and get the ` access_token `. Pass it in the headers and you will be able to access the user list endpoint `http://127.0.0.1:8000/api/v1/users/`

The header prefix should be `Bearer `
```
Bearer <Token>
```

Sample response on login
```
{
    "user": {
        "username": "mwinel",
        "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1NTU1NzkwNTksImlhdCI6MTU1NTU3NzU1OSwic3ViIjp7InVzZXJuYW1lIjoibXdpbmVsIiwicGFzc3dvcmQiOiJwcmluY2U2MTgifX0.VSy0rCGeM32ndfuGF8RuYiQc055r9ibaxPCvBs0LxF8"
    }
}
```

### **What needs to be done?**
There is need to write more unit tests and cover user signup validations.

### **Relevant user stories**
[#165369465](https://www.pivotaltracker.com/story/show/165369465)
